src/Uno.UI.Runtime.Skia.AppleUIKit/UI/Xaml/Controls/TextBox/InvisibleTextBoxViewExtension.cs

```diff
@@ -222,39 +228,39 @@ internal void ProcessNativeTextInput(string? text)
 
 	private string? GetNativeText() => _textBoxView?.Text;
 
-	private IInvisibleTextBoxView CreateNativeView(TextBox textBox) => _owner?.TextBox?.AcceptsReturn != true ?
-		new SinglelineInvisibleTextBoxView(this) : new MultilineInvisibleTextBoxView(this);
+	private IInvisibleTextBoxView CreateNativeView()
+	{
+		return _owner?.TextBox?.AcceptsReturn != true ?
+			new SinglelineInvisibleTextBoxView(this) :
+			new MultilineInvisibleTextBoxView(this);
+	}
 
-	public void AddViewToTextInputLayer(XamlRoot xamlRoot)
+	private void AddViewToTextInputLayer(XamlRoot xamlRoot)
 	{
 		if (_textBoxView is not UIView nativeView)
 		{
 			return;
 		}
 
-		if (GetOverlayLayer(xamlRoot) is { } layer && nativeView.Superview != layer)
+		if (GetHost(xamlRoot) is { } host && GetOverlayLayer(xamlRoot) is { } layer && nativeView.Superview != layer)
 		{
-			layer.AddSubview(nativeView);
+			host.AddViewToTextInputLayer(nativeView);
 
-			// Push the overlay native view out of the visible view - this way
-			// the blue typing suggestion overlay will not be shown to the user.
 			nativeView.Frame = new CoreGraphics.CGRect(-1000, -1000, 10, 10);
 		}
 	}
 
-	public void RemoveViewFromTextInputLayer()
+	private void RemoveViewFromTextInputLayer(XamlRoot xamlRoot)
 	{
-		if (_textBoxView is not UIView nativeView)
+		if (GetHost(xamlRoot) is { } host)
 		{
-			return;
-		}
-
-		if (nativeView.Superview is not null)
-		{
-			nativeView.RemoveFromSuperview();
+			host.RemoveLatestViewFromTextInputLayer();
```

Comment:
Instead of having to add this method, maybe it would be simpler (not sure though, just a thought) to check if there is already a view from previous focus, store that view in a variable, add the new one, and then explicitly remove the old view.

Reply:
Doing this logic on the same file felt weird, although I see it can be better in the long run. I wanted to keep it centralized, but I will update the PR with your suggestion removing the public methods on the RootViewController